src/main/java/com/snowflake/kafka/connector/internal/streaming/channel/ExposingInternalsTopicPartitionChannel.java

```diff
@@ -0,0 +1,58 @@
+package com.snowflake.kafka.connector.internal.streaming.channel;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.snowflake.kafka.connector.internal.streaming.BufferedTopicPartitionChannel;
+import com.snowflake.kafka.connector.internal.streaming.telemetry.SnowflakeTelemetryChannelStatus;
+import com.snowflake.kafka.connector.internal.telemetry.SnowflakeTelemetryService;
+import dev.failsafe.Fallback;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestChannel;
+
+public interface ExposingInternalsTopicPartitionChannel {
```

Comment:
To avoid confusion, the intention was to create this interface as a midstep before investing time in further refactoring. These methods existed already in the previous implementation and were freely accessible by other classes (used to have `protected` modifier).

In the future, I'd like to get rid of the interface but for now, I'd like to focus on discussing whether having a single interface `TopicPartitionChannel` having 2 implementations (buffered and non-buffered) is the way to go.

Additional refactoring will be done afterward.